Dockerfile

```
@@ -12,34 +12,26 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.21.3@sha256:02d7116222536a5cf0fcf631f90b507758b669648e0f20186d2dc94a9b419a9b AS builder
+FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_9_1.21@sha256:98a0ff138c536eee98704d6909699ad5d0725a20573e2c510a60ef462b45cce0 AS builder
 ENV APP_ROOT=/opt/app-root
 ENV GOPATH=$APP_ROOT
```

Im guessing you switched your Add to the copy because you were having issues with it. To the best of my knowledge this is because the `brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_9_1.21@sha256:98a0ff138c536eee98704d6909699ad5d0725a20573e2c510a60ef462b45cce0` image does not have an `/opt/app-root` path. If you create that dir and the src dir inside that it should work fine. 
```suggestion
ENV GOPATH=$APP_ROOT
RUN mkdir /opt/app-root && mkdir /opt/app-root/src
```

---

Perfect thanks for pointing this out ive been swapping between different dockerfiles and must've gotten confused.